plugins/adapters/modbus/srcCxx/ModbusAdapterConfiguration.cxx

```diff
@@ -443,43 +444,48 @@ std::vector<long double> ModbusAdapterConfigurationElement::get_float_value(
             if (dynamic_data::is_signed_kind(element_kind)) {
                 int8_t value = static_cast<int8_t>(input[i]);
                 check_correct_value(
-                        static_cast<long double>(static_cast<int8_t>(value)),
+                        static_cast<long double>(value),
                         index,
                         element_kind);
 
                 float_vector.push_back(
-                        (int8_t) data_offset() + value * data_factor());
+                        rti::utils::long_double::safe_cast<int8_t>(
+                                data_offset() + value * data_factor()));
             } else {
                 uint8_t value = static_cast<uint8_t>(input[i]);
                 check_correct_value(
-                        static_cast<long double>(static_cast<uint8_t>(value)),
+                        static_cast<long double>(value),
                         index,
                         element_kind);
 
                 float_vector.push_back(
-                        (unsigned) (int8_t) data_offset()
-                        + value * data_factor());
+                        rti::utils::long_double::safe_cast<uint8_t>(
+                                data_offset() + value * data_factor()));
             }
             break;
         }
         case ModbusDataType::holding_register_int16:
         case ModbusDataType::input_register_int16: {
             if (dynamic_data::is_signed_kind(element_kind)) {
                 int16_t value = input[i];
-                check_correct_value(value, index, element_kind);
+                check_correct_value(
+                        static_cast<long double>(value),
```

Comment:
Why not use the `safe_cast` here?

Reply:
the `safe_cast` just defines the casting from a `long double` to another type, it does not define the casting from another type to `long double` because that cast should be safe by using `static_cast`